src/index.ts

```diff
@@ -27,15 +28,53 @@ const createGoError = <E extends Error>(err: unknown): GoResultError<E> => {
   return fail(new Error('' + err));
 };
 
-export const go = async <T, E extends Error>(fn: Promise<T> | (() => Promise<T>)): Promise<GoResult<T, E>> => {
+export const go = async <T, E extends Error>(
+  fn: Promise<T> | (() => Promise<T>),
+  options?: PromiseOptions
+): Promise<GoResult<T, E>> => {
+  const attemptOptions: AttemptOptions<any> = {
+    delay: options?.retryDelayMs || 200,
+    maxAttempts: options?.retries || 1,
+    initialDelay: 0,
+    minDelay: 0,
+    maxDelay: 0,
+    factor: 0,
+    timeout: options?.timeoutMs || 0,
+    jitter: false,
+    handleError: null,
+    handleTimeout: options?.timeoutMs
```

Comment:
We could handle the retry + timeout options case through the library's `handleTimeout` function like this.

The difference will be that if options.timeoutMs is defined and it ends up retrying the maximum times and still fails, it will throw an `Operation timed out after final retry` instead of re-throwing the original error (if any).

Is this more like what you had in mind?

Reply:
Yes and no :smile: 

I definitely don't want to use [exceptions for a flow control](https://softwareengineering.stackexchange.com/a/189225). Would it be problematic to call the whole function again from within the `handleTimeout` handler? I mean the `go` function. I think the flow ends after the `handleTimeout` anyway, so it shouldn't be a problem. From the documentation:
```
The handleTimeout function should return a Promise that will be the return value of the retry() function.
```
So, the result will be eventually returned. I think that can work.

Reply:
Yes it is possible to call `go` inside `handleTimeout`, but it seems to work a bit differently (see [here](https://github.com/api3dao/promise-utils/pull/5/files#diff-258035e5968f6bf645400d417f310218d7d9a9a10606a3c34e7f55db193f58f3R115)).

Or do you see another way to use it? I think we can also remove of `retryFn` entirely in this case as well.

Reply:
I can't commit a failing test, but I had this kind of a scenario:
```js
it('retries and resolves timed out functions', async () => {
    const res = await retryTimeoutGo(
      () =>
        new Promise((res) =>
          setTimeout(() => {
            res(2);
          }, 50)
        ),
      { timeoutMs: 50, retries: 3 }
    );
    expect(res).toEqual(fail(new Error('Operation timed out after final retry')));
  });
```

In the end this fails with the following result which doesn't seem right:
```js
{
  data: {
    data: {
      error: [Error: Operation timed out after final retry],
      success: false,
    },
    success: true,
  },
  success: true,
}
```

However, this test passed with the `retryFn` (because if was handled through the exception).

Reply:
This makes perfect sense and works for that case as well, Will add